lte/gateway/c/session_manager/test/BUILD.bazel

```diff
@@ -0,0 +1,280 @@
+# Copyright 2021 The Magma Authors.
+
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+load("@rules_cc//cc:defs.bzl", "cc_test")
+
+test_suite(
+    name = "non_flaky_test",
+    tags = ["-flaky"],
```

Comment:
Flaky-ness at Google was established using the cc_test `flaky` directive documented [here](https://docs.bazel.build/versions/main/be/common-definitions.html#test.flaky). `If set, executes the test up to three times, marking it as failed only if it fails each time. By default, this attribute is set to False and the test is executed only once. Note, that use of this attribute is generally discouraged - tests should pass reliably when their assertions are upheld.`

I would say that if a unit test is very likely to pass with three attempts (per cc_test `flaky` flag above) we should include it in CI.  If not, then maybe a test suite is merited... or we should discard an exceptionally flaky test =D Or fix it.

With this test_suite approach, how does this work? Will Bazel execute all non-flaky test suites across the `bazel test <targets>` so long as the tag is identical? How do we selectively add unit tests in this bazel target?

If we go with this approach, any opinion on `-flaky` versus `-noflakes` or something? With `-flaky` I could see this being either "only run flaky tests" or "no flaky by virtue of the minus sign in front". Not totally clear to the user. WDYT?

Reply:
I completely missed the actual flaky attribute! The running three times is very useful. Will try that out!

Reply:
resolved - now we use the built in flaky tag

Reply:
would definitely add a comment. I looked at `-flaky` and thought it meant we were asking bazel to pass in a flag and "only run flaky tests"

Reply:
I think in her new PR (maybe yet unpushed?) This whole test_suite thing is gone and @themarwhal is using Bazel's cc_test builtin for flaky annotations (so no command line flag necessary - will run flaky annotated tests three tries for success).

Reply:
Yes, in the updated PR I've removed the `non_flaky_test` target as we might not need it. The built in flaky config that I've added to some of the tests will indicate to Bazel to always run it up to 3 times before indicating failure.